### ISSUE TITLE
`Type.Omit` mixed with `Type.Union` is not preserving the `additionalProperties` attribute

### ISSUE THREAD

**Issue by jlaloi** [sinclairzx81/typebox#944]
`Type.Omit` mixed with `Type.Union` **with multiple schemas** is not preserving the `additionalProperties` attribute.

For example with:

``` ts
import { Type } from '@sinclair/typebox'

const userSchema = Type.Object(
  {
    id: Type.String(),
    firstName: Type.String(),
  },
  { additionalProperties: false }
)

const dogSchema = Type.Object(
  {
    id: Type.String(),
    name: Type.String(),
  },
  { additionalProperties: false }
}
```

 ❌ `additionalProperties` will NOT be preserved with:
```ts

Type.Omit(Type.Union([userSchema, dogSchema]), ['id'])

 {
  anyOf: [
    {
      type: 'object',
      properties: {
        firstName: { type: 'string', [Symbol(TypeBox.Kind)]: 'String' }
      },
      required: [ 'firstName', [length]: 1 ],
      [Symbol(TypeBox.Kind)]: 'Object'
    },
    {
      type: 'object',
      properties: { name: { type: 'string', [Symbol(TypeBox.Kind)]: 'String' } },
      required: [ 'name', [length]: 1 ],
      [Symbol(TypeBox.Kind)]: 'Object'
    },
    [length]: 2
  ],
  [Symbol(TypeBox.Kind)]: 'Union'
}
```

 ✅ `additionalProperties` wil be preserved with only one schema in the `Type.Union`:
```ts
Type.Omit(Type.Union([userSchema]), ['id'])

{
  additionalProperties: false,
  type: 'object',
  properties: { firstName: { type: 'string', [Symbol(TypeBox.Kind)]: 'String' } },
  required: [ 'firstName', [length]: 1 ],
  [Symbol(TypeBox.Kind)]: 'Object'
}
```

 ✅ `additionalProperties` will be preserved with
```ts
Type.Union([Type.Omit(userSchema, ['id']), Type.Omit(dogSchema, ['id'])])

{
  anyOf: [
    {
      additionalProperties: false,
      type: 'object',
      properties: {
        firstName: { type: 'string', [Symbol(TypeBox.Kind)]: 'String' }
      },
      required: [ 'firstName', [length]: 1 ],
      [Symbol(TypeBox.Kind)]: 'Object'
    },
    {
      additionalProperties: false,
      type: 'object',
      properties: { name: { type: 'string', [Symbol(TypeBox.Kind)]: 'String' } },
      required: [ 'name', [length]: 1 ],
      [Symbol(TypeBox.Kind)]: 'Object'
    },
    [length]: 2
  ],
  [Symbol(TypeBox.Kind)]: 'Union'
}
```

Thanks for your dedication on your amazing library :)

**Comment by sinclairzx81** [comment]
@jlaloi Hi, thanks for reporting.

Fix should be available on 0.33.2. 

Cheers
S

**Comment by jlaloi** [comment]
Thanks for the speedy fix! Really appreciated.
